app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  before_save :downcase_email
+  validates :name, presence: true, length: {maximum: 50}
+  validates :email, presence: true, length: {maximum: 255},
+    format: {with: Settings.email_regex},
+    uniqueness: true
+  validates :password, presence: true, length: {minimum: 6}
+  has_secure_password
```

Comment:
dòng này đùng để làm gì em ơi?

Reply:
Dòng đó sẽ thêm các chức năng như là:
+ lưu password đã được hash vào csdl
+ tạo ra password and password_confirmation và xác thực password_confirmation phải giống với password
+ tạo ra method authenticate để ktra mật khẩu của 1 user